File: my-folder/0128-longest-consecutive-sequence/solution.ts
function longestConsecutive(nums: number[]): number {
  const set = new Set(nums);
  let longest = 0;
  for (const n of set) {
    if (!set.has(n - 1)) {
      let length = 1;
      while (set.has(n + length)) {
        length += 1;
      }
      longest = Math.max(longest, length);
    }
  }
  return longest;
}

File: my-folder/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        stored = set(nums)

        for n in stored:
            if n - 1 not in stored:
                length = 0
                while n + length in stored:
                    length += 1
                longest = max(longest, length)

        return longest


